### DISCUSSION TITLE
Reading/converting from trimmed nurbs

### DISCUSSION THREAD

**Discussion by jobordu** [gismo/gismo#581] [Q&A]
Hi, 

I am new with g+smo, I am still discovering the library. 

I tried reading a .3dm geometry with trimmed nurbs, but it seems that all the trimmed nurbs weren't converted to IGA, there were instead represented as **un**trimmed nurbs.

It was my understanding that g+smo would convert a trimmed nurbs to an IGA representation when using gsview.exe. 

Is there something I am doing wrong ? What can I do to convert a trimmed nurbs to an IGA ?

**Comment by jobordu** [comment]
@filiatra @stefantakacs

Any new developments concerning this task ? My goal is to import a trimmed-nurns STEP file or OpenNURBS file, apply multiples punctual constraints on the geometry and solve using a biharmonic equation. 

If there are developments, Is there a demo or tutorial ? 

Best,

**Comment by hverhelst** [comment]
Hi @jobordu , I have some experience with opennurbs input from Rhino and indeed saw that trimming is not handled. I don't think this will happen in the near future. @filiatra what do you think?
